llama/model.py

```diff
@@ -69,11 +72,12 @@ def apply_rotary_emb(
     xk: torch.Tensor,
     freqs_cis: torch.Tensor,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
-    xq_ = torch.view_as_complex(xq.float().reshape(*xq.shape[:-1], -1, 2))
```

Comment:
@Liyang90 @AlexWertheim Here is the change to rotary embedding logic

Reply:
The first dimension of the `freqs_cis` corresponds to the input sequence length dimension, which is the second dimension of `xq`. These two dimension needs to be aligned. In our cases it's 1, but it is not the assumption. 
`freqs_cis` shape can be [L, h_dim/2], and `xq` shape can be [BS, L, n_h, h_dim].

Reply:
I found each row is selected from the pre-generated `freqs_cis` in each forward pass https://github.com/pytorch-tpu/llama/blob/3881ba25178b67e80167ff412471e2c51e76abfb/llama/model.py#L296. So I found the first dimension is always 1 in this function.

Reply:
Synced with Liyang offline, for the current usage input seqlen is always 1, we should support arbitrary seqlen here.

Reply:
LGTM. Just make sure that `apply_rotary_emb` is tested and is producing exact same output with this change.

Reply:
Yes, it's tested in a colab. https://colab.research.google.com/gist/lsy323/79d9d50801300db41033c47503f10338/rotary_embedding-test.ipynb